### PR TITLE
PIMOB-2183: Update KtLint version and .editorconfig & run ktlintFormat

### PR DESCRIPTION
**Pull request by jheng-hao-lin-cko**
## Issue

[PIMOB-2183](https://checkout.atlassian.net/browse/PIMOB-2183)

## Proposed changes

* Update KtLint version to `0.50.0`
* Update `.editorconfig` to align with other Android repos
* Run `ktlintFormat` to autoformat against the `.editorconfig`
* Rename `showAlertDialog` to `Extensions` to fit the ktlint style

## Test Steps

Run `ktlint` locally and pass. No functionality changed.

## Checklist

* [x] Reviewers assigned
* [x] I have performed a self-review of my code and manual testing
* [x] Lint and unit tests pass locally with my changes

[PIMOB-2183]: https://checkout.atlassian.net/browse/PIMOB-2183?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
+ij_kotlin_allow_trailing_comma = true
+ij_kotlin_allow_trailing_comma_on_call_site = true
 
-# According to https://kotlinlang.org/docs/reference/coding-conventions.html#names-for-test-methods it is acceptable to write method names
-# in natural language. When using natural language, the description tends to be longer. Allow lines containing an identifier between
-# backticks to be longer than the maximum line length. (Since 0.41.0)
 [**/test/**.kt]
-ktlint_ignore_back_ticked_identifier = true
\ No newline at end of file
+max_line_length = unset
```
Comment (chintan-soni-cko) [inline .editorconfig:15]: It ignores the max line validation in test cases. Can we ignore max line length validation for test names only?

File: frames/src/test/java/com/checkout/frames/screen/billingaddressdetails/usecase/BillingAddressInputComponentStyleUseCaseTest.kt [hunk 2]
```diff
@@ -84,16 +88,16 @@ internal class BillingAddressInputComponentStyleUseCaseTest {
                     infoStyle = TextLabelViewStyle(),
                     inputFieldStyle = InputFieldViewStyle(),
                     subtitleStyle = TextLabelViewStyle(),
-                    titleStyle = TextLabelViewStyle()
-                )
+                    titleStyle = TextLabelViewStyle(),
+                ),
 
-            )
+                )
```
Comment (chintan-soni-cko) [inline frames/src/test/java/com/checkout/frames/screen/billingaddressdetails/usecase/BillingAddressInputComponentStyleUseCaseTest.kt:94]: Empty space, worth adding a check for it

Comment (jheng-hao-lin-cko) [reply frames/src/test/java/com/checkout/frames/screen/billingaddressdetails/usecase/BillingAddressInputComponentStyleUseCaseTest.kt]: Hmmm looks like we don't have this option :(

**Review by chintan-soni-cko — APPROVED** [review]
LGTM

**Pull request merged**